src/core/coap/coap.cpp

```diff
@@ -704,7 +704,7 @@ void CoapBase::ProcessReceivedRequest(Message &aMessage, const Ip6::MessageInfo
             *curUriPath++ = '/';
         }
 
-        VerifyOrExit(curUriPath + optionLength < OT_ARRAY_END(uriPath), OT_NOOP);
+        VerifyOrExit(curUriPath + optionLength + 1 < OT_ARRAY_END(uriPath), error = OT_ERROR_PARSE);
```

Comment:
I don't follow why we would need `+1` here in the `VerifyOrExit` check. Can you please explain? Thanks.

Here is my reading of the code (maybe I am missing something)?
- `curUriPath` points to current location to write to.
- When reading the value we would write `optionLength` chars from `curUriPath[0]` up to `curUriPath[optionLength-1]`
- And update the `curUriPath` by `curUriPath += optionLength` 
- The check `curUriPath + optionLength  < OT_ARRAY_END(uriPath)` ensures the updated `curUriPath`  still points to somewhere within the array (not passed the end of array). 
- And we should still be able to do `*curUriPath = '\0';` (since `curUriPath` should still point to a valid byte within array).

Reply:
I think you are right. 
I changed this PR to just introduce two minor enhancements. 
Thanks.